units/player.gd

```diff
@@ -30,8 +58,5 @@ func _physics_process(delta):
 		position.x += HORIZONTAL_SPEED * delta
 
 	if Input.is_action_just_pressed("fire"):
-		print('fired!')
-		var p = Projectile.instantiate()
-		owner.add_child(p)
-		p.transform = projectile_spawner.global_transform
+		fire()
```

Comment:
:wine_glass: :cheese: 

Unnecessary change ultimately, but was done in an early stage of this PR implementation that played with the idea of projectile effects calling `fire` directly on the player scene. That didn't work the way I wanted it to, but I decided to keep the fire mechanism as its own function because it _feels_ right, and if we ever want to do a little more with this mechanism we'd probably have to do this refactor anyway.

Reply:
Eventually we'll want to add some state machine here to tidy this all up 👍 meantime this is a nice prefactor :D